src/components/ThemeSwitcher.svelte

```diff
@@ -0,0 +1,75 @@
+<script>
+  import Fa from 'svelte-fa/src/fa.svelte'
+  import FaLayers from 'svelte-fa/src/fa-layers.svelte'
+  import { onMount } from 'svelte'
+  import { theme } from '../store/theme'
+  import { faMoon } from '@fortawesome/free-solid-svg-icons/faMoon'
+  import { faSun } from '@fortawesome/free-solid-svg-icons/faSun'
+
+
+  function toggleTheme() {
+    theme.update((current) => {
+      return current === 'light' ? 'dark' : 'light'
+    })
+
+    theme.subscribe((value) => {
+      localStorage.setItem('theme', value)
+    })
+  }
+
+  onMount(() => {
+    let currentTheme = localStorage.getItem('theme') ?? 'light'
```

Comment:
If there is no `localStorage.getItem('theme')`, we can also fallback to the user agent preferences: https://stackoverflow.com/a/57795495/3440266